### ISSUE TITLE
RangeError: Maximum call stack size exceeded.

### ISSUE THREAD

**Issue by Gigabyte-Giant** [Gigabyte-Giant/SameSyntax_Template#1]
In the demo program, clicking the canvas results in a RangeError.

**Steps to reproduce:**
In any browser, open "demos/simple/processingTemplate.htm".
Open the Javascript console.
Click canvas.

**Comment by Gigabyte-Giant** [comment]
...as it appears, performing any action that would fire an event (clicking, pressing keys, etc.), results in this problem.
I'm guessing it's my code.

**Comment by Gigabyte-Giant** [comment]
Just a link to reference:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError

The error occurs when PJS' goes to bind my code, to their functions, example:
```javascript
$p.mouseReleased = mouseReleased;
mouseReleased = mouseReleased.bind($p);;
function mouseClicked() {
if (typeof mouseClicked !== 'undefined' && mouseClicked) {
mouseClicked();
}
}```

Perhaps separating the user code from the execution code would be something to do? I doubt it would fix it, but, it might be something to look into.

**Comment by Gigabyte-Giant** [comment]
The issue was, in fact, caused by my environment variables (function aliases specifically), being mixed with the PJS code. I fixed this in (34d64fe)[https://github.com/Gigabyte-Giant/ProcessingTemplate/commit/34d64fe02fda0383506ccf332494b5309ad1a455].

The only downside to my fix, is user code must now be placed within a "with-statement" (`with (<something>) { <some code> }`). However, we are still able to run Khan Academy code, without any syntactical changes!

After a bit more testing, I'll close the issue if all succeeds.

**Comment by Gigabyte-Giant** [comment]
Basic tests succeeded! (see test code [here](https://github.com/Gigabyte-Giant/SameSyntax_Template/tree/master/tests/basic_test))

Issue fixed in 34d64fe. Closing.
